### DISCUSSION TITLE
Use Jest for testing

### DISCUSSION THREAD

**Discussion by ThatXliner** [atom/atom#22517] [Ideas]
Not only it's multi-threaded, it seems to have better documentation, etc! A lot of people already use jest

**Comment by icecream17** [comment]
That reminds me of an idea I had the other day to use TypeScript, which also has a bonus transpiler

That way new Javascript features can be used

**Reply by icecream17** [reply]
The babel 7 pr is closed now. 

But it seems like there's massive codebase improvement potential....

**Reply by icecream17** [reply]
Forgot to mention: https://github.com/atom/atom/issues/22455
